prometheusx/metrics.go

```diff
@@ -1,32 +1,70 @@
 package prometheus
 
 import (
+	"net/http"
+	"strconv"
+
 	"github.com/pkg/errors"
 	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+	"github.com/urfave/negroni"
 )
 
 // Metrics prototypes
 type Metrics struct {
-	ResponseTime *prometheus.HistogramVec
+	responseTime    *prometheus.HistogramVec
+	totalRequests   *prometheus.CounterVec
+	duration        *prometheus.HistogramVec
+	responseSize    *prometheus.HistogramVec
+	requestSize     *prometheus.HistogramVec
+	handlerStatuses *prometheus.CounterVec
 }
 
 // Method for creation new custom Prometheus  metrics
 func NewMetrics(app, version, hash, date string) *Metrics {
+	labels := map[string]string{
+		"app":       app,
+		"version":   version,
+		"hash":      hash,
+		"buildTime": date,
+	}
 	pm := &Metrics{
-		ResponseTime: prometheus.NewHistogramVec(
+		responseTime: prometheus.NewHistogramVec(
 			prometheus.HistogramOpts{
-				Name: app + "_response_time_seconds",
-				Help: "Description",
-				ConstLabels: map[string]string{
-					"version":   version,
-					"hash":      hash,
-					"buildTime": date,
-				},
+				Name:        "response_time_seconds",
+				Help:        "Description",
+				ConstLabels: labels,
 			},
 			[]string{"endpoint"},
 		),
+		totalRequests: prometheus.NewCounterVec(prometheus.CounterOpts{
+			Name:        "requests_total",
```

Comment:
Since some services support different API protocols (REST, gRPC) I think these metrics should be prefixed with `http_` or (for gRPC) `grpc_`

Reply:
Hello, sorry for my late response. I'm not an expert about gRPC, but could you give me an example wherein Kratos (or another ory service) we are using this instrumentation? Because I think that for the gRPC handler there should be different metrics manager as for example statuses are different, and probably logic for handlers itself?

Reply:
Hey there, sorry for my late reply too :) We are using gRPC in Ory Keto and are planning on adding it to more services at some point. It is true that not everything overlaps (e.g. status codes), but some things do overlap like requests_total. I saw in the prometheus docs and somewhere else that prefixing the protocol is a good idea, which is why I was asking about this :)

Reply:
Will 8610e55 suffice?

Reply:
Absolutely! :)